ixmp/backend/base.py

```diff
@@ -192,6 +193,95 @@ def set_unit(self, name, comment):
         get_units
         """
 
+    def read_file(self, path, item_type: ItemType, filters, **kwargs):
+        """OPTIONAL: Read Platform, TimeSeries, or Scenario data from file.
+
+        A backend **may** implement read_file for one or more combinations of
+        the `path` and `item_type` methods. For all other combinations, it
+        **must** raise :class:`NotImplementedError`.
+
+        Parameters
+        ----------
+        path : os.PathLike
+            File for input. The filename suffix determines the input format:
+
+            ====== ===
+            Suffix Format
+            ====== ===
+            .csv   Comma-separated values
+            .gdx   GAMS data exchange
+            .xlsx  Microsoft Office Open XML spreadsheet
+            ====== ===
+
+        item_type : ItemType
+            Type(s) of items to read.
+        filters : dict of dict of str
```

Comment:
@khaeru is it `dict of dict of str`?

what do you think about moving `filters` for both methods to `kwargs`? (as it seems to be optional parameter in some cases)

Reply:
`dict (str -> list of str)`—sorry, that's my mistake.

Reply:
> moving `filters` for both methods to `kwargs`

Yes, that's a good idea. Nice catch! I think `item_type` needs to say as it very much changes how the backend handles the function call.